packages/plantae/src/ky/createKyHooks.spec.ts

```diff
@@ -0,0 +1,230 @@
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [headerSetPlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    const result = await kyWithHoks
+      .get("https://example.com/api/v1/foo")
+      .text();
+
+    expect(result).toStrictEqual("request plugin is activated");
+  });
+
+  test("method", async () => {
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [postMethodPlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    const result = await kyWithHoks
+      .get("https://example.com/api/v1/foo")
+      .text();
+
+    expect(result).toEqual("post request is completed");
+  });
+
+  test("body", async () => {
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [postMethodWithBodyPlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    const result = await kyWithHoks
+      .post("https://example.com/api/v1/bar")
+      .json();
+
+    expect(result).toStrictEqual({ foo: "bar" });
+  });
+
+  test("url", async () => {
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [modifyUrlPlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    const result = await kyWithHoks
+      .get("https://example.com/api/v1/foo")
+      .text();
+
+    expect(result).toEqual("url is modified");
+  });
+
+  test.skip("signal", async () => {
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [abortSignalPlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    await expect(
+      kyWithHoks.get("https://example.com/api/v1/delayed")
+    ).rejects.toThrow();
+  });
+});
+
+describe("ky:afterResponse -", () => {
+  test("headers", async () => {
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [modifiedHeaderResponsePlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    const res = await kyWithHoks.get("https://example.com/api/v1/foo");
+
+    expect(res.headers.get("x-foo")).toStrictEqual("bar");
+  });
+
+  // FIXME: SyntaxError: Unexpected token b in JSON at position 0
+  test.skip("body", async () => {
+    const hooks = createKyHooks({
+      client: ky,
+      plugins: [modifiedResponseBodyPlugin()],
+    });
+
+    const kyWithHoks = ky.extend({
+      hooks,
+    });
+
+    const result = await kyWithHoks
+      .get("https://example.com/api/v1/foo")
+      .json();
+
+    expect(result).toStrictEqual("baz");
+  });
+
+  // FIXME: It should be checked about the spec
+  test.skip("overriden plugin", async () => {
+    const firstHooks = createKyHooks({
+      client: ky,
+      plugins: [firstPlugin()],
+    });
+
+    const firstKyWithHoks = ky.extend({
+      hooks: firstHooks,
+    });
+
+    const secondHooks = createKyHooks({
+      client: firstKyWithHoks,
+      plugins: [secondPlugin()],
+    });
+
+    const secondKyWithHooks = ky.extend({
```

Comment:
```suggestion
    const secondKyWithHooks = firstKyWithHoks.extend({
```

Please extend `firstKyWithHoks`, not `ky` directly.

Reply:
@orionmiz 

I'd like to point out that the `createKyHooks` function, which is declared second, should receive `firstKyWithHooks` as an argument, rather than `ky` like below;

```  
const secondHooks = createKyHooks({
  client: firstKyWithHoks, // NOTE: Is it still valid? I couldn't verify that.
  plugins: [secondPlugin()],
});
 ```

Reply:
The `client` parameter (meaning `firstKyWithHoks` for here) for adapters is only used for `retry` argument in plugin interafces.

Reply:
So it's valid.